File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    do
    {
        printf("Change owed:");
        change = get_float();
    }

    //Converting floating point dollars to integer cents
    while (change < 0);
    int cents = round(change * 100);

    int coins = 0;

    //Can quarters be used?
    while (cents >= 25)
    {
        cents = cents - 25;
        coins++;
    }

    //Can dimes be used?
    while (cents >= 10)
    {
        cents = cents - 10;
        coins++;
    }

    //Can nickels be used?
    while (cents >= 5)
    {
        cents = cents - 5;
        coins++;
    }

    //Can pennies be used?
    while (cents >= 1)
    {
        cents = cents - 1;
        coins++;
    }
    printf("%i\n", coins);
}

File: cash.py
from cs50 import get_float

# reprompting user until input is a positive numerical value
while True:
    print("change owed: ", end="")
    change = get_float()

    if change > 0:
        break

# converting change from dollars to cents
cents = int(change * 100)

# number of whole quarters that can go into change value
quarters = cents // 25
cents = cents % 25

# number of whole dimes that can go into remaining change value
dimes = cents // 10
cents = cents % 10

# number of whole nickels that can go into remaining change value
nickels = cents // 5
cents = cents % 5

pennies = cents

coins = quarters + dimes + nickels + pennies
print(coins)
